### PR TITLE
Webpack: webpack applied

### PR DESCRIPTION
**Pull request by Shaharyar-saleem**

### CODE REVIEW

File: assets/js/blockchain/constants.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const BUSD/**/ = {
```
Comment (bobachef) [inline assets/js/blockchain/constants.js:1]: remove this extra comment

File: assets/js/blockchain/busdContract.js [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
 const PAZ_ADDRESS = "0xb9f5C03d733C9F7eBA6B81c48A55ef0bEBBB200A";
 
-const reinvestContractAddress = {
-  56: "0xE8064A1C0CeC63D15dC0b4338bCB2D5d4740876c",
-  97: "0xf7cDEbf75f830bf83C4BC6C005b437D7298cE2d7",
-};
+const reinvestContractAddress = constants.reinvestContractAddress;
 const busdAbi = require('./abi/busd.js');
-// const busdContractAbi = [
+let chainID;
```
Comment (cchen408) [inline assets/js/blockchain/busdContract.js:10]: `chainId` is the proper convention

Comment (bobachef) [inline assets/js/blockchain/busdContract.js:10]: `chainId` is the proper convention

File: assets/js/blockchain/connectWallet.js [hunk 3]
```diff
@@ -2,9 +2,10 @@ const Web3Modal = window.Web3Modal.default;
 const WalletConnectProvider = window.WalletConnectProvider.default;
 const Fortmatic = window.Fortmatic;
 const evmChains = window.evmChains;
-const userData = require('./pazziveContract');
+
 const reinvestData = require('./reinvestContract');
```
Comment (bobachef) [inline assets/js/blockchain/connectWallet.js:6]: this should be `reinvest`

File: assets/js/blockchain/connectWallet.js [hunk 4]
```diff
@@ -2,9 +2,10 @@ const Web3Modal = window.Web3Modal.default;
...
 const busdData = require('./busdContract');
```
Comment (bobachef) [inline assets/js/blockchain/connectWallet.js:7]: this should be `busd`

File: assets/js/blockchain/connectWallet.js [hunk 5]
```diff
@@ -2,9 +2,10 @@ const Web3Modal = window.Web3Modal.default;
...
+const userData = require('./pazziveContract');
```
Comment (bobachef) [inline assets/js/blockchain/connectWallet.js:8]: this should be `paz`

File: assets/js/blockchain/connectWallet.js [hunk 6]
```diff
@@ -2,9 +2,10 @@ const Web3Modal = window.Web3Modal.default;
...
 const trackerData = require('./dividendsTrackerContract');
```
Comment (bobachef) [inline assets/js/blockchain/connectWallet.js:9]: this should be `tracker`

File: assets/js/blockchain/constants.js [hunk 7]
```diff
@@ -17,9 +17,35 @@ const TRACKER = {
...
 
+const reinvestContractAddress = {
+  56: "0xE8064A1C0CeC63D15dC0b4338bCB2D5d4740876c",
+  97: "0xf7cDEbf75f830bf83C4BC6C005b437D7298cE2d7",
+};
+
+
+const etherProvider = new ethers.providers.Web3Provider(window.ethereum);
+const signer = etherProvider.getSigner();
+let chainID = 56;
```
Comment (bobachef) [inline assets/js/blockchain/constants.js:28]: i don't think this belongs here.. especially if it's a `let`

File: assets/js/blockchain/constants.js [hunk 8]
```diff
@@ -17,9 +17,35 @@ const TRACKER = {
...
+const PAZ_DECIMALS = 18;
+
+async function getChainId(){
+  return await signer.getChainId();
```
Comment (bobachef) [inline assets/js/blockchain/constants.js:32]: this doesn't belong here this should go in a helper or utility file

File: assets/js/blockchain/constants.js [hunk 9]
```diff
@@ -17,9 +17,35 @@ const TRACKER = {
...
+}
+
+async function getUserAddress(){
```
Comment (bobachef) [inline assets/js/blockchain/constants.js:35]: this doesn't belong here either.. the only thing that should be in here are constants.

File: assets/js/blockchain/reinvestContract.js [hunk 10]
```diff
@@ -208,16 +211,18 @@ const reinvestContractAbi = [
     type: "function",
   },
 ];
-
+let chainID;
```
Comment (bobachef) [inline assets/js/blockchain/reinvestContract.js:214]: `chainId` is the convention

File: assets/js/blockchain/pazziveContract.js [hunk 11]
```diff
@@ -1,9 +1,11 @@
 const etherProvider = new ethers.providers.Web3Provider(window.ethereum);
-const signer = etherProvider.getSigner();
+const constants = require('./constants');
+const signer = constants.signer;
+const reinvestData = require('./reinvestContract');
 const BURNT_ADDRESS = "0x000000000000000000000000000000000000dEaD";
 const TEST_CONTRACT = "0x0B036007dbbeAd58dA9469140E3B0a8E7e7bC4c6";
 const LIVE_CONTRACT = "0xb9f5C03d733C9F7eBA6B81c48A55ef0bEBBB200A";
```
Comment (bobachef) [inline assets/js/blockchain/pazziveContract.js:7]: all of these should be constants and should just be referenced from there

**Review by bobachef — CHANGES_REQUESTED** [review]
some feedback

File: assets/js/blockchain/constants.js (hunk 1 above)
Comment (Shaharyar-saleem) [reply assets/js/blockchain/constants.js]: removed

File: assets/js/blockchain/busdContract.js (hunk 2 above)
Comment (Shaharyar-saleem) [reply assets/js/blockchain/busdContract.js:10]: followed

File: assets/js/blockchain/connectWallet.js (hunk 3 above)
Comment (Shaharyar-saleem) [reply assets/js/blockchain/connectWallet.js:6]: followed

File: assets/js/blockchain/connectWallet.js (hunk 6 above)
Comment (Shaharyar-saleem) [reply assets/js/blockchain/connectWallet.js:9]: followed

File: assets/js/blockchain/busdContract.js [hunk 12]
```diff
@@ -1,288 +1,30 @@
...
+const reinvestContractAddress = constants.reinvestContractAddress;
+let chainId;
+
+async function getChainId(){
+  return await signer.getChainId();
+}
+
+async function getUserAddress(){
+  return await signer.getAddress();
+}
```
Comment (bobachef) [inline assets/js/blockchain/busdContract.js:15]: this is a bit redundant imo what we can always try is

```
const signer = constants.signer;
const {getChainId, getAddress:getUserAddress} = signer; 

await getChainId()
await getUserAddress()

```

this is called object destructuring.

Comment (Shaharyar-saleem) [reply assets/js/blockchain/busdContract.js:15]: done

**Review by bobachef — APPROVED** [review]
lgtm

**Pull request merged**